backend/src/socket/socket.service.ts

```diff
@@ -0,0 +1,157 @@
+import { Server, Socket } from 'socket.io';
+import { Injectable } from '@nestjs/common';
+import { AuthService } from '../auth/auth.service';
+import { UserService } from '../users/user/user.service';
+import {
+	ChatId,
+	ChatList,
+	Chat_List_Item,
+	MessageList,
+	SingleMessage,
+	SocketMessage,
+} from './socket.types';
+
+@Injectable()
+export class SocketService {
+	constructor(
+		private readonly authService: AuthService,
+		private readonly userService: UserService,
+	) {}
+	public chatServer: Server = null;
+
+	public chatListSubscribers: ChatList = {};
+	public messageLists: MessageList = {};
+
+	chatIdToRoomName(chatId: string): string;
+	chatIdToRoomName(chatId: ChatId): string;
+	chatIdToRoomName(chatId: ChatId | string): string {
+		return `room_${chatId}`;
+	}
+
+	async joinRoom(chatId: ChatId, socket: Socket) {
+		const cookie: string = this.jwtCookieFromHandshakeString(
+			socket.handshake.headers.cookie,
+		);
+		try {
+			const userId: number = this.authService.userIdFromCookieString(cookie);
+			await this.userService.findOne({ where: { id: userId } });
```

Comment:
correct. It's quite hard to fake a user ID in a valid JWT token (since it uses a secret from our .env), so as an "attack vector" it's not really an issue here. But if for example a user would be deleted and then comes back this would catch that user not existing any more.